kubetest/kops.go

```diff
@@ -179,12 +215,38 @@ func newKops(provider, gcpProject, cluster string) (*kops, error) {
 	if err := os.Setenv("AWS_SSH_KEY", sshKey); err != nil {
 		return nil, err
 	}
-	// ZONE is required by the AWS e2e tests.
-	zones := strings.Split(*kopsZones, ",")
+
+	// zones are required by the kops e2e tests.
+	var zones []string
+
+	// if zones is set to zero and gcp project is not set then pick random aws zone
+	if *kopsZones == "" && provider == "aws" {
+		// Initialize global pseudo random generator.
+		// We do not need to intialize it if kopsZones is passed in or we are running the other tests.
+		// That is why I did not put it in main func.
+		rand.Seed(time.Now().UnixNano())
+
+		zones, err = getRandomAWSZones(*kopsMasterCount)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		zones = strings.Split(*kopsZones, ",")
+	}
+
+	// set ZONES for e2e.go
 	if err := os.Setenv("ZONE", zones[0]); err != nil {
 		return nil, err
 	}
 
+	if len(zones) == 0 {
+		return nil, errors.New("no zones found")
+	} else if zones[0] == "" {
+		return nil, errors.New("zone cannot be a empty string")
```

Comment:
Can we validate all the zones?

Reply:
A really good idea, but do you mind if we do it in another PR?  I am wondering if that work is outside the scope of this PR.

Reply:
Also, my plan is to not pass in zones long term.  Always pick a random zone or zones with the new func.  Because of this zones would be chosen programmatically.